### PR TITLE
Move the applyThousandSeparator method to utils

### LINKED ISSUE #4566: Move the applyThousandSeparator method to Utils
### Description

In #4527, we introduced a new method (`applyThousandSeparator`) to add a thousand separator to the transactions number.

The method should be moved to `client/transactions/utils/utils.js` and then we can import the function wherever needed. This will keep the logic separate and we will also be able to add tests to the method.

### Acceptance criteria

* Move the `applyThousandSeparator` method to `client/transactions/utils/utils.js`
* Import the method to `client/transactions/list/index.tsx`

### PR DESCRIPTION
**Pull request by fashuvo**
Fixes #4566

#### Changes proposed in this Pull Request

In #4527, we used the `applyThousandSeparator` method to add a thousand separator to the transactions number. Later we found out that we need to add the separator to the deposits and disputes too.

This PR moves the `applyThousandSeparator` method to `client/utils/index.js` and then imports the method to use in the transactions, deposits, and disputes counts. This will keep the logic separate and we will also be able to add tests to the method.

#### Testing instructions

1. Clone the `update/move-applyThousandSeparator-method-to-utils` branch in your local machine
2. Navigate to `woocommerce-payments/client/transactions/list/index.tsx`
3. Run `npm run watch` in the terminal
4. Find this line in the index.tsx file: `applyThousandSeparator(transactionsSummary.count as number)`
5. Change `transactionsSummary.count` to any number greater than 999 and save changes
6. Refresh `wp-admin > Payments > Transactions` and you should see the thousand separator in the transaction count.
7. Deposits can be tested in the similar way from `woocommerce-payments/client/deposits/list/index.tsx`
8. And disputes can be tested in the similar way from `woocommerce-payments/client/disputes/index.tsx`

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

File: changelog/update-move-applyThousandSeparator-method-to-client-utils [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Significance: patch
```
Comment (deepakpathania) [inline changelog/update-move-applyThousandSeparator-method-to-client-utils:1]: Let's keep a single changelog file for a PR and update the changes there.

**Review by deepakpathania — COMMENTED** [review]
Changes LGTM, just left a small comment.

File: changelog/update-move-applyThousandSeparator-method-to-client-utils (hunk 1 above)
Comment (fashuvo) [reply changelog/update-move-applyThousandSeparator-method-to-client-utils:1]: Thanks. I have removed the previous changelog entry.

**Review by deepakpathania — APPROVED** [review]
LGTM ✅

**Pull request merged**